codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -109,12 +124,23 @@ public void incrementWindowSize(ChannelHandlerContext ctx, Http2Stream stream, i
     @Override
     public void consumeBytes(ChannelHandlerContext ctx, Http2Stream stream, int numBytes)
             throws Http2Exception {
-        state(stream).consumeBytes(ctx, numBytes);
+        if (stream.id() == CONNECTION_STREAM_ID) {
+            throw new UnsupportedOperationException("Returning bytes for the connection window is not supported");
+        }
+        if (numBytes <= 0) {
+            throw new IllegalArgumentException("numBytes must be positive");
+        }
+
+        // Streams automatically consume all remaining bytes when they are closed, so just ignore
+        // if already closed.
+        if (!isClosed(stream)) {
+            state(stream).consumeBytes(ctx, numBytes);
+        }
     }
 
     @Override
     public int unconsumedBytes(Http2Stream stream) {
-        return state(stream).unconsumedBytes();
+        return isClosed(stream) ? 0 : state(stream).unconsumedBytes();
```

Comment:
We probably don't need to do this in this PR. This was the result of some of my work for adding stream history, where closed streams would no longer have a `FlowState` property.  Since we still need to apply flow control for closed streams, I didn't want these methods to require that the `FlowState` exist.  WDYT ... revert this for now?